### PR TITLE
sstable/block: add comment about restart points and numRestarts

### PR DESCRIPTION
**Pull request by itsbilal**
Add a comment that makes reading / understanding this code and
the interactions between offsets and restarts easier.

### CODE REVIEW

**Comment by petermattis** [comment]
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/pebble/416)

File: sstable/block.go [hunk 1]
```diff
@@ -120,10 +120,28 @@ type blockEntry struct {
...
+	// A "restart point" in a block is a point where the full key is encoded,
+	// instead of just having a suffix of the key encoded. See readEntry() for
+	// how prefix compression of keys works. Keys in between two restart points
+	// only have a suffix encoded in the block. When restart interval is 1, no
+	// prefix compression of keys happens. This is the case with range tombstone
+	// blocks.
+	//
+	// All restart offsets are listed in increasing order in
+	// i.ptr[i.restarts:len(block)-4], while numRestarts is encoded in the last
+	// 4 bytes of the block as a Uint32 (i.ptr[len(block)-4:]). i.restarts can
```
Comment (petermattis) [inline sstable/block.go]: Nit: `s/Uint32/uint32/g`

File: sstable/block.go [hunk 2]
```diff
@@ -120,10 +120,28 @@ type blockEntry struct {
...
+	// therefore be seen as the point where data in the block ends, and a list
+	// of offsets of all restart points begins.
+	restarts int32
+	// Number of restart points in this block. Encoded at the end of the block
+	// as a Uint32.
```
Comment (petermattis) [inline sstable/block.go]: Nit: `s/Uint32/uint32/g`

**Comment by itsbilal** [comment]
Addressed comments. Thanks for the review!

**Pull request merged**